apps/address-service/domains/common/utils/addressKeyUtils.js

```diff
@@ -0,0 +1,46 @@
+const get = require('lodash/get')
+
+const JOINER = '~'
+const SPACE_REPLACER = '_'
+
+/**
+ * @param {NormalizedBuilding} normalizedBuilding
+ * @returns {string}
+ */
+function generateAddressKey (normalizedBuilding) {
+    const data = normalizedBuilding.data
+    /**
+     * @type {string[]}
+     */
+    const parts = [
+        get(data, 'country'),
+        get(data, 'region'),
+        get(data, 'area'),
+        get(data, 'city'),
+        get(data, 'city_district'),
+        get(data, 'settlement'),
+        get(data, 'street'),
+        get(data, 'house'),
+        get(data, 'block'),
+    ]
+
+    return parts
+        // Remove empty parts
+        .filter(Boolean)
+        // Keep single space between words
+        .map(
```

Comment:
What about filtering some punctuation characters and other, that is not belonging to words?

Reply:
Done